packages/realm/index.react-native.js

```diff
@@ -19,4 +19,19 @@
 /* eslint-disable @typescript-eslint/no-var-requires -- We're exporting using CJS assignment */
 /* eslint-env commonjs */
 
-module.exports = require("./dist/bundle.react-native").Realm;
+// eslint-disable-next-line no-undef -- In React Native, process is not defined, but in Jest it is
+const isJest = process?.env?.JEST_WORKER_ID !== undefined;
```

Comment:
I'm a bit sad that we need to do this 😞 It would be better if we could find a way to keep this pure from runtime checks of the environment ... historically we've been bitten by the brittleness of these many times before.

Reply:
We can get around the `react-native` preset for `jest`, which is basically ensuring the `react-native` entry point is being loaded.  I did a lot of digging around this and this was the best solution I could find.  At least it's very dependent on Jest and has very little impact on the experience.

Reply:
I need to refine the comment. `process` is actually defined in `react-native`, but for some reason it is not typed.

Reply:
> which is basically ensuring the react-native entry point is [being loaded](https://github.com/facebook/react-native/blob/main/packages/react-native/jest/react-native-env.js)

As per that link, it looks to me that we could probably export the Node.js bundle via the `require` condition and avoid the runtime check?

Reply:
Add this line after https://github.com/realm/realm-js/blob/0b9dd8b31371d6cd4fe9e39bf06022b219756415/packages/realm/package.json#L31

```
      "require": "./index.node.js",
```